channel.go

```diff
@@ -418,6 +419,7 @@ It is safe to call this method multiple times.
 
 */
 func (ch *Channel) Close() error {
+	defer ch.connection.releaseChannel(ch.id)
```

Comment:
Sorry, forgive my poor English. I didn't get your point. Here I use a defer, I'm sure it will release after connection received the `channel.close-ok`, it will prevent  `channel.open` for the same channel number.